featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -53,50 +55,53 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event: Dict[str, Optional[str]] = {}
-    if not evaluation_event.feature:
+
+    feature = evaluation_event.feature
+
+    if not feature:
         return
-    event[FEATURE_NAME] = evaluation_event.feature.name
-    event[ENABLED] = str(evaluation_event.enabled)
-    event["Version"] = EVALUATION_EVENT_VERSION
+
+    event: Dict[str, Optional[str]] = {
+        FEATURE_NAME: feature.name,
+        ENABLED: str(evaluation_event.enabled),
+        "Version": EVALUATION_EVENT_VERSION,
+    }
+
+    reason = evaluation_event.reason
+    variant = evaluation_event.variant
 
     # VariantAllocationPercentage
-    if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
-        if evaluation_event.variant:
-            event[VARIANT] = evaluation_event.variant.name
-        event[REASON] = evaluation_event.reason.value
-
-        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
-            allocation_percentage = 0
-
-            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
-                for allocation in evaluation_event.feature.allocation.percentile:
-                    if (
-                        evaluation_event.variant
-                        and allocation.variant == evaluation_event.variant.name
-                        and allocation.percentile_to
-                    ):
-                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+    if reason and reason != VariantAssignmentReason.NONE:
```

Comment:
This if should be removed. We plan to always emit the reason- even when it's None. The following lines should work without this if.

Reply:
Did you make this same comment twice, or am I missing something?

Reply:
Yeah- UI bugged out on me and I thought I lost this comment